### PR TITLE
fixing tests

### PR DESCRIPTION
**Pull request by sah**

### CODE REVIEW

File: isign/signer.py [hunk 1]
```diff
@@ -158,29 +158,31 @@ class HashEntry(asn1crypto.core.Sequence):
...
-                elif parsed[0].native == SHA256_OID:
-                    for cd_hash in cd_hashes:
-                        if cd_hash['hashType'] == SHA256_HASHTYPE:
-                            val = cd_hash[SHA256_HASHTYPE]
-                else:
-                    raise ValueError('unexpected entry %s' % parsed[0].native)
-                signer_info['signed_attrs'][3][1][i] = asn1crypto.core.Any(
-                    HashEntry({"ident": parsed[0],
-                               "value": asn1crypto.core.OctetString(val)}))
+            if len(signer_info['signed_attrs']) > 3:
```
Comment (neilk) [inline isign/signer.py]: so the idea here is that this is a telltale sign of the old-style stuff

File: isign/signer.py [hunk 2]
```diff
@@ -158,29 +158,31 @@ class HashEntry(asn1crypto.core.Sequence):
...
+                        HashEntry({"ident": parsed[0],
+                                   "value": asn1crypto.core.OctetString(val)}))
 
             # Update plist of truncated CodeDirectory hashes
-            plist = plistlib.readPlistFromString(signer_info['signed_attrs'][4][1][0].native)
-            plist['cdhashes'] = [plistlib.Data(cd_hash[cd_hash['hashType']][:20])
-                                 for cd_hash in cd_hashes]
-            signer_info['signed_attrs'][4][1][0] = asn1crypto.core.Any(
-                asn1crypto.core.OctetString(plistlib.writePlistToString(plist)))
+            if len(signer_info['signed_attrs']) > 4:
```
Comment (neilk) [inline isign/signer.py]: and another telltale sign

File: isign/archive.py [hunk 3]
```diff
@@ -189,7 +189,7 @@ def precheck(cls, path):
                 if plist_path not in zipfile_obj.namelist():
                     return False
                 plist = cls.get_info(relative_bundle_dir, zipfile_obj)
-        is_native = IosApp.is_native(plist)
```
Comment (neilk) [inline isign/archive.py]: this entire precheck is dubious, it only delays the failure

I think I wrote this b/c we had so many other other uploads to sift through at that other company

**Review by neilk — APPROVED** [review]
LGTM but can we add some comments where I mentioned?

File: isign/signer.py (hunk 1 above)
Comment (sah) [reply isign/signer.py]: Not just a telltale sign, but also we're about to rewrite a new field that, for old-style stuff, doesn't exist. So, this says "only rewrite it if it exists".

A good comment might be something like "Older signatures lack some fields. For now, we don't attempt to add them if they're not already there."

File: isign/signer.py (hunk 2 above)
Comment (neilk) [reply isign/signer.py]: it looks like you changed to 4 b/c you added one attr above. If that's the case can we combine both if clauses? Or are they still separate?

**Pull request merged**